### PR TITLE
[MOS-326] Change call logic removed from call window

### PR DESCRIPTION
**Pull request by kkleczkowski**
Call logic is now removed from call window and call app.
There was spaghetti logic mixed in window and app, now
call logic is based on notificatins from service cellular.

**Description**

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

File: module-apps/application-call/ApplicationCall.cpp [hunk 1]
```diff
@@ -82,34 +82,45 @@ namespace app
...
         });
         addActionReceiver(manager::actions::AbortCall, [this](auto &&data) {
-            callerIdTimer.reset();
             if (const auto state = getState(); state == Application::State::ACTIVE_FORGROUND) {
-                switchWindow(window::name_call, std::make_unique<app::CallAbortData>());
+                switchWindow(window::name_call);
             }
             else if (state == Application::State::ACTIVE_BACKGROUND) {
-                setCallState(call::State::IDLE);
+                callModel->setState(app::call::CallState::None);
```
Comment (pholat) [inline module-apps/application-call/ApplicationCall.cpp:89]: to completely follow we should set the model first -> then based on presenter reaction switch window instead? I'm not requesting it. There always has to be some limit on how much we want to change code for the better at once and I firmly believe that this PR met the requirements and above :)

Comment (pholat) [inline module-apps/application-call/ApplicationCall.cpp:89]: to completely follow we should set the model first -> then based on presenter reaction switch window instead? I'm not requesting it. There always has to be some limit on how much we want to change code for the better at once and I firmly believe that this PR met the requirements and above :)

File: module-apps/application-call/ApplicationCall.cpp [hunk 2]
```diff
@@ -82,34 +82,45 @@ namespace app
...
         addActionReceiver(manager::actions::HandleCallerId, [this](auto &&data) {
             auto callParams = static_cast<app::manager::actions::CallParams *>(data.get());
-            handleCallerId(callParams);
+            callModel->setPhoneNumber(callParams->getNumber());
+            callModel->setState(call::CallState::Incoming);
+            auto window = getCurrentWindow();
+            if (window->getName() != app::window::name_call) {
+                LOG_INFO("Switch to call window");
+                switchWindow(app::window::name_call);
+            }
```
Comment (pholat) [inline module-apps/application-call/ApplicationCall.cpp:119]: same as here: https://github.com/mudita/MuditaOS/pull/3745/files#r851120600

Comment (pholat) [inline module-apps/application-call/ApplicationCall.cpp:119]: same as here: https://github.com/mudita/MuditaOS/pull/3745/files#r851120600

File: module-apps/application-call/include/application-call/ApplicationCall.hpp [hunk 3]
```diff
@@ -17,7 +17,7 @@
 namespace app
 {
     inline constexpr auto name_call       = "ApplicationCall";
-    inline constexpr auto call_stack_size = 8192U;
+    inline constexpr auto call_stack_size = 8192 * 2U;
```
Comment (pholat) [inline module-apps/application-call/include/application-call/ApplicationCall.hpp:20]: please remember to double-check the stack requirements

Comment (pholat) [inline module-apps/application-call/include/application-call/ApplicationCall.hpp:20]: please remember to double-check the stack requirements

File: module-apps/application-call/model/CallModel.cpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        else {
+            callState = newState;
+        }
+
+        if (callState == CallState::Active) {
+            turnOnKeypadBacklight();
+        }
+        else {
+            turnOffKeypadBacklight();
+        }
```
Comment (pholat) [inline module-apps/application-call/model/CallModel.cpp:50]: while setting the model state is ok - turning stuff on and off less so ( I know that this is mostly moved logic here and do not request to change it all - I'm just mentioning it as I go on with the review :) )

Comment (pholat) [inline module-apps/application-call/model/CallModel.cpp:50]: while setting the model state is ok - turning stuff on and off less so ( I know that this is mostly moved logic here and do not request to change it all - I'm just mentioning it as I go on with the review :) )

File: module-apps/application-call/windows/CallWindow.hpp [hunk 5]
```diff
@@ -74,10 +57,17 @@ namespace gui
...
+        void refreshWindow() override;
+
+        void updateDuration(const UTF8 &text, bool isVisible = true) override;
+        void setNavBarForActiveCall() override;
+        void setNavBarForIncomingCall() override;
+        void clearNavBar() override;
+        void showIncomingCallLayout(bool isValidCallerId) override;
+        void showActiveCallLayout() override;
+        void showCallEndedLayout() override;
+        void updateNumber(const UTF8 &text) override;
```
Comment (pholat) [inline module-apps/application-call/windows/CallWindow.hpp:70]: I personally prefer to have functions in the view named `showSomething` as `set` or `update` assumes that there was a previous state - and the view shouldn't mind state too much. 
Then we i.e. have flow like that:
`window.onInput() -> presenter.handleUIEnter() -> ui.showNumber()`
which might be a bit easier to read as we immediately know that UI only shows what the presenter handled before. ( I hope it was explanatory enough)

Comment (pholat) [inline module-apps/application-call/windows/CallWindow.hpp:70]: I personally prefer to have functions in the view named `showSomething` as `set` or `update` assumes that there was a previous state - and the view shouldn't mind state too much. 
Then we i.e. have flow like that:
`window.onInput() -> presenter.handleUIEnter() -> ui.showNumber()`
which might be a bit easier to read as we immediately know that UI only shows what the presenter handled before. ( I hope it was explanatory enough)

File: module-services/service-cellular/service-cellular/CellularMessage.hpp [hunk 6]
```diff
@@ -602,17 +602,17 @@ class CellularAnswerIncomingCallMessage : public CellularMessage
...
   public:
     CellularHangupCallMessage() : CellularMessage(Type::HangupCall)
     {}
 
-    [[nodiscard]] auto toAction() const -> std::unique_ptr<app::manager::ActionRequest>
-    {
-        return std::make_unique<app::manager::ActionRequest>(
-            sender, app::manager::actions::AbortCall, std::make_unique<app::manager::actions::ActionParams>());
-    }
+    //    [[nodiscard]] auto toAction() const -> std::unique_ptr<app::manager::ActionRequest>
```
Comment (pholat) [inline module-services/service-cellular/service-cellular/CellularMessage.hpp:611]: jsust remove :)

Comment (pholat) [inline module-services/service-cellular/service-cellular/CellularMessage.hpp:611]: jsust remove :)

**Review by pholat — APPROVED** [review]
1. Please read through the comments and select if you want to improve the code more now. In my opinion, it's in much better shape and the goal seems to be met.
2. Feel free to close comments upon reading - these are for you, treat them as checkpoints.

PR approved seems legit, thanks for the UT and overall clean aproach it's apreciated, especially in applictaion we did not revisit for a long time.

File: module-apps/application-call/model/CallModel.cpp (hunk 4 above)
Comment (kkleczkowski) [reply module-apps/application-call/model/CallModel.cpp:50]: to be honest - it should be implemented in event manager. It's only moved logic

File: module-apps/application-call/include/application-call/ApplicationCall.hpp (hunk 3 above)
Comment (kkleczkowski) [reply module-apps/application-call/include/application-call/ApplicationCall.hpp]: done

Comment (kkleczkowski) [reply module-apps/application-call/include/application-call/ApplicationCall.hpp]: done

File: module-apps/application-call/ApplicationCall.cpp [hunk 7]
```diff
@@ -145,6 +156,29 @@ namespace app
...
+            callModel->setState(app::call::CallState::Ended);
+            return sys::MessageNone{};
+        });
+
+        connect(typeid(cellular::CallStartedNotification), [&](sys::Message *request) {
+            auto message = static_cast<cellular::CallStartedNotification *>(request);
+            callModel->setPhoneNumber(message->getNumber());
+            callModel->setState(app::call::CallState::Outgoing);
+            return sys::MessageNone{};
+        });
```
Comment (mjan-dev) [inline module-apps/application-call/ApplicationCall.cpp:180]: :heart:

**Review by mjan-dev — APPROVED** [review]
LGTM, great work! :+1: Some of the logic should be moved (as @pholat), however I suppose it is topic for another task this change is already big one.

**Pull request merged**